pkg/services/alerting/conditions/query.go

```diff
@@ -294,3 +295,63 @@ func validateToValue(to string) error {
 	_, err := time.ParseDuration(to)
 	return err
 }
+
+// FrameToSeriesSlice converts a frame that is a valid time series as per data.TimeSeriesSchema()
+// to a TimeSeriesSlice.
+func FrameToSeriesSlice(frame *data.Frame) (tsdb.TimeSeriesSlice, error) {
+	tsSchema := frame.TimeSeriesSchema()
+	if tsSchema.Type == data.TimeSeriesTypeNot {
+		// If no fields, or only a time field, create an empty tsdb. with a single
+		// time series in order to trigger "no data" in alerting.
+		if len(frame.Fields) == 0 || (len(frame.Fields) == 1 && frame.Fields[0].Type().Time()) {
+			return tsdb.TimeSeriesSlice{{
+				Name:   frame.Name,
+				Points: make(tsdb.TimeSeriesPoints, 0),
+			}}, nil
+		}
+		return nil, fmt.Errorf("input frame is not recognized as a time series")
+	}
+
+	seriesCount := len(tsSchema.ValueIndices)
+	seriesSlice := make(tsdb.TimeSeriesSlice, 0, seriesCount)
+	timeField := frame.Fields[tsSchema.TimeIndex]
+	timeNullFloatSlice := make([]null.Float, timeField.Len())
+
+	for i := 0; i < timeField.Len(); i++ { // built slice of time as epoch ms in null floats
+		tStamp, err := timeField.FloatAt(i)
+		if err != nil {
+			return nil, err
+		}
+		timeNullFloatSlice[i] = null.FloatFrom(tStamp)
+	}
+
+	for _, fieldIdx := range tsSchema.ValueIndices { // create a TimeSeries for each value Field
+		field := frame.Fields[fieldIdx]
+		ts := &tsdb.TimeSeries{
+			Name:   field.Name,
+			Points: make(tsdb.TimeSeriesPoints, field.Len()),
+		}
+
+		if field.Labels != nil {
+			ts.Tags = field.Labels.Copy()
+			// Tags are appended to the name so they are eventually included in EvalMatch's Metric property
+			// for display in notifications.
+			ts.Name = fmt.Sprintf("%v {%v}", ts.Name, field.Labels.String())
```

Comment:
I guess the idea is to have a fallback on name when not provided. But some data sources sets the name and/or display name and I wonder whether we only should fallback if there are no existing name/display name. Or maybe I'm missing something?

For example, see azure monitor if an alias is provided or not: https://github.com/grafana/grafana/blob/54c503dcc3d930a1bcb74d3c3293d41971e2a245/pkg/tsdb/azuremonitor/azuremonitor-datasource.go#L294-L312

Reply:
@marefr Good call. I have updated it to use the displayName if it is set.

When using the Alias query field in the Azure Monitor service in the Azure Monitor data source, it sets the field config's display name:

Without Alias:

![image](https://user-images.githubusercontent.com/1692624/95488374-75efe380-0963-11eb-83c9-9a59502727cb.png)

With an Alias ( e.g. `🧵 {{blobtype}}  🧵 {{tier}}`):

![image](https://user-images.githubusercontent.com/1692624/95488529-a9cb0900-0963-11eb-9bb5-1c73cf9a3d8e.png)

![image](https://user-images.githubusercontent.com/1692624/95488576-b5b6cb00-0963-11eb-9c72-d8ed1429d4a3.png)